frontend/components/player/now_playing_info.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect } from "react";
+
+const NowPlayingInfo = ({
+	audioRef,
+	track,
+	trackProgress,
+	isPlaying,
+	updateTrackProgress,
+}) => {
+	if (!track) {
+		track = {
+			title: "",
+			albumId: "",
+			audioUrl: "",
+			albumImageUrl: "",
+			songArtist: "",
+		};
+	}
+	const { title, albumId, audioUrl, albumImageUrl, songArtist } = track;
+	useEffect(() => {
+		updateTrackProgress(audioRef.current.currentTime);
+	}, [isPlaying]);
+	return (
+		<div className="now-playing">
+			{trackProgress > 0 || (isPlaying && trackProgress === 0) ? (
```

Comment:
Do you have a suggestion? This compound conditional helped me fix bugs I was having relevant to the NowPlayingInfo box.

1st case is needed when a user is playing a song, but then pauses the song. This allows the track info to stay in the box even though `isPlaying` is no longer true.
2nd case is needed because when the info box receives a new track, `isPlaying` is true when the component re-renders but at that moment, trackProgress is still 0. And trackProgress doesn't trigger a re-render, so without the second case, the box had stayed empty even when the song began playing.

Let me know your thoughts.

Reply:
@k-shan I will merge this PR but curious whether you had a suggestion for simplifying. Thanks!